Number isNaN polyfill

Ikkyu-os opened the pull request:
PR for https://github.com/Financial-Times/polyfill-service/issues/522

Tested on several IE & FF versions.

triblondon commented on polyfills/Number/isNaN/tests.js:
```diff
@@ -0,0 +1,43 @@
+	it('numbers', function () {
+		expect(Number.isNaN(37)).to.be(false);
+	});
+	it('numbers', function () {
```
looks like these tests have some copy/paste remnants that need sorting out.

triblondon commented on polyfills/Number/isNaN/config.json:
```diff
@@ -0,0 +1,12 @@
+		"chrome":"<19",
+		"safari": "<16"
+	},
+	"docs": "https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN"
```
Also add a `spec` link if possible.

triblondon commented:
Thanks for this.  A few comments for you to deal with.  I am working on a proper contributor agreement, but in the meantime would you please confirm your agreement to the following:

* This code is all your own work
* You are publishing it under an MIT licence

Thanks.

Ikkyu-os commented:
Hello

To answer your questions:

- The code is my own, although the polyfill itself comes from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isNaN . Not sure if its an issue, as there is not 100 way to implement it anyway :)
- I publish it under the MIT licence, I updated the config file to reflect this.

I updated the test to remove useless code as suggested and add all browsers according to current support.

triblondon commented on polyfills/Number/isNaN/tests.js:
```diff
@@ -0,0 +1,27 @@
+});
+
+describe('returns false with', function () {
+	it('numbers', function () {
```
incorrectly described tests.  I suggest either drop the `describe`s and call the nested test cases "returns true for NaNs" and "returns false for valid numbers and non-number data types"; or create an `it` for each value, eg returns false with... string, undefined, empty object.

Ikkyu-os commented:
Ok. I didn't understood how they were wrongly described, thanks for the explanation. 

I'll update the tests likely today.

triblondon commented:
There's a dodgy indent now, but I'll let you off that :-).  Thanks so much for the contribution!
